src/arch/riscv/interrupts.c

```diff
@@ -43,7 +50,11 @@ void interrupts_arch_init()
 
 void interrupts_arch_ipi_send(cpuid_t target_cpu, irqid_t ipi_id)
 {
-    sbi_send_ipi(1ULL << target_cpu, 0);
+    if(platform.arch.aclint_base == 0) {
+        sbi_send_ipi(1ULL << target_cpu, 0);
+    } else {
+        aclint_send_ipi(target_cpu);
+    }
```

Comment:
Could we try to make this compile-time constant? Maybe using the platform script to generate a `CLINT_PRESENT` macro based on `platform.arch.aclint_base`...